### ISSUE TITLE
process_group.block -> call_block -> eval_lang -> eval -> eval

### ISSUE THREAD

**Issue by T0J0** [yihui/knitr#1101]
Hi there,

I don't know what happened but my previously working Knitr code does not compile anymore. And that, even though I cannot remember to have changed anything (does RStudio do automatic package updates???).

Anyway, I have an issue setting a size variable in the "setup" chunk as was proposed [here](http://stackoverflow.com/a/29671036/3162788). 

In order to track the error in my code, I took the minimal example which is provided in the link, and added a chunk which sources local r scripts of mine. A subsequent chunk is supposed to create a plot out of data which was created by the scripts. If I do not refer to one of the size variables in my figure chunk, the PDF file compiles without any problem (although the figure chunks of the minimal example already feature the size variables). However, if I try to refer to one of the size variables in my figure chunk, the PDF file does not compile (e.g., `fig.height = 0.2*a4height`) and gives the error (translated from German):

```
error in eval(expr, envir, enclos) : object 'a4height' not found
Calls: knit ... process_group.block -> call_block -> eval_lang -> eval -> eval
```

It took me already hours to narrow the issue down to this. Unfortunately, I was not able to reproduce the error with a common, minimal example so far. The following is the structure of my code:

```
\documentclass[12pt,a4paper]{article}
\usepackage[utf8]{inputenc}
\usepackage[english]{babel}
\usepackage[T1]{fontenc}
\usepackage{graphicx}
\usepackage{tikz}
<<setup, include=FALSE, cache=FALSE>>=
  library(knitr)
  options(formatR.arrow=TRUE,width=50)
  opts_chunk$set(fig.path='test/graphics-', 
                 cache.path='cacheTest/graphics-', 
                 fig.align='center',
                 dev='tikz',
                 external=TRUE,
                 echo=FALSE,
                 cache=FALSE,
                 par=TRUE
                )
  txtWidth <- 6.1023622
  a4height<- 11.7
@

\begin{document}
\begin{figure}[!ht]
\centering
<<fig_1, fig.width = txtWidth, fig.height = 0.2*a4height>>=
    x = seq(1,3,l=100)
    plot(x,cos(x), type ="l", xlab = "x", ylab = "y")
@

<<fig_2, fig.width = 1*txtWidth, fig.height = 0.5*a4height>>=
    x = seq(1,3,l=100)
    plot(x,cos(x), type ="l", xlab = "x", ylab = "y")
@
\caption{Test figure}
\end{figure}

% Until here from minimal example
% http://stackoverflow.com/a/29671036/3162788

<<load-r, warning = FALSE, message = TRUE>>=

# Sourcing some files
source("getData.R", encoding = "UTF-8-BOM")
source("...", encoding = "UTF-8-BOM")
source("...", encoding =  "UTF-8-BOM")

# and manipulate the data
...
...

# Test print
print(5 +5)
@

<<plotTest, echo=FALSE, warning = FALSE, message = FALSE, fig.height = 0.2*a4height>>= 

# Data preparation
...
...

# And plot
boxplot(..., data = get(callData))
@

\end{document}
```

Any help appreciated a lot!

**Comment by yihui** [comment]
If you are not able to make a reproducible example, I'm afraid I cannot help you either. My best guess is you cleared the global environment somewhere in your code, hence the object `a4height` was removed.

**Comment by T0J0** [comment]
Thanks a lot for coming back to me! Your guess was indeed right.

I included `rm(list = ls())` at the beginning of the "load-r" chunk. I don't know why I removed it from the above example. I think I have had some wrong concept. I thought the size variables are stored in another environment then the r code.

However, which still keeps me puzzled: Why did it work previously? Might caching be an explanation? But even if the size variable was cached, the subsequent chunk removed all variables from the environment's workspace.
